src/pudl/output/ferc714.py

```diff
@@ -652,18 +652,20 @@ def summarized_demand_ferc714(
             demand_hourly_pa_ferc714.loc[
                 :, ["report_date", "respondent_id_ferc714", "demand_mwh"]
             ],
+            on=["report_date", "respondent_id_ferc714"],
             how="left",
         )
-        .groupby(["report_date", "respondent_id_ferc714"])
-        .agg({"demand_mwh": sum})
+        .groupby(["report_date", "respondent_id_ferc714"], as_index=False)[
+            ["demand_mwh"]
+        ]
+        .sum(min_count=1)
         .rename(columns={"demand_mwh": "demand_annual_mwh"})
-        .reset_index()
         .merge(
             georeferenced_counties_ferc714.groupby(
-                ["report_date", "respondent_id_ferc714"]
-            )
-            .agg({"population": sum, "area_km2": sum})
-            .reset_index()
+                ["report_date", "respondent_id_ferc714"], as_index=False
+            )[["population", "area_km2"]].sum(min_count=1),
+            on=["report_date", "respondent_id_ferc714"],
+            how="left",
```

Comment:
the left merge here is the real substantive change.

everything else is speed/clarity updates:
* `groupby.agg` is slow so I converted them all to `groupby.sum`
* `sum(min_count=1)` means that we don't get a bunch of 0's and then subsequently `inf`s in the following assign
* added `as_index=False` into the `groupby`s so we don't need to keep resetting the index
* added `merge` `on` columns